src/main/java/com/epam/grid/engine/cmd/CommandArgUtils.java

```diff
@@ -21,84 +21,85 @@
     public static String[] splitCommandIntoArgs(final String command) {
-        final String[] arr = checkAndMergeQuotesStrings(StringUtils.tokenizeToStringArray(command, DELIMITERS));
-        return Arrays.stream(arr).map(String::trim).filter(s -> !s.isEmpty()).toArray(String[]::new);
-    }
-
-    /**
-     * This method searches for parts of quoted strings that are split into several strings when parsing
-     * and combines them into one string.
-     *
-     * @param commandParts The command from the template after parsing.
-     * @return The command's structure.
-     */
-    private static String[] checkAndMergeQuotesStrings(final String[] commandParts) {
         final List<String> result = new ArrayList<>();
-        final StringBuilder quotedPart = new StringBuilder();
+        final StringBuilder token = new StringBuilder();
+
+        boolean isQuote = false;
+        boolean isToken = false;
+        boolean isQuoteEscapedToken = false;
 
-        for (String commandPart : commandParts) {
-            if (commandPart.startsWith(QUOTE) || commandPart.startsWith(APOSTROPHE)) {
-                appendCommandToBuilder(quotedPart, commandPart);
-            } else if (commandPart.endsWith(QUOTE) || commandPart.endsWith(APOSTROPHE)) {
-                appendCommandToBuilder(quotedPart, commandPart);
-                dumpQuotedToResults(quotedPart, result);
-            } else {
-                addQuotedPartToBuilderOrDump(commandPart, quotedPart, result);
+        for (int i = 0; i < command.length(); i++) {
+            if (!isToken) {
+                if (isWhitespaceCharacter(command.charAt(i))) {
+                    continue;
+                }
+                isToken = true;
             }
-        }
-        dumpQuotedToResultsIfPresent(quotedPart, result);
-        return result.toArray(new String[0]);
-    }
 
-    private static void addQuotedPartToBuilderOrDump(final String commandPart, final StringBuilder quotedPart,
-                                                     final List<String> result) {
-        if (quotedPart.length() != 0 && !quotedPart.toString().endsWith(QUOTE)
-                && !quotedPart.toString().endsWith(APOSTROPHE)) {
-            appendCommandToBuilder(quotedPart, commandPart);
-        } else {
-            dumpQuotedToResultsIfPresent(quotedPart, result);
-            result.add(commandPart);
-        }
-    }
+            if (!isQuote && isWhitespaceCharacter(command.charAt(i))) {
+                result.add(token.toString());
+                token.setLength(0);
+                isToken = false;
+                continue;
+            }
 
-    private static void dumpQuotedToResults(final StringBuilder quotedPart, final List<String> result) {
-        result.add(quotedPart.toString());
-        quotedPart.setLength(0);
-    }
+            if (command.charAt(i) == QUOTE) {
+                if (token.length() == 1 && command.charAt(i - 1) == BACKSLASH) { //a start of token with escaped quote
```

Comment:
Am I correct that it will work only for case
```
"\"some text in quotes\""
```
but not for
```
"some_text:\"some text in quotes\""
```

Reply:
yes, this part is only for such tokens that are fully enclosed in escaped quotes

Reply:
so is it possible to change that? I believe this approach could be slightly changed to work with such case too

Reply:
Actually I am wondering do we need to worry about `mirrored` quotes? So we can check that current quote is a `mirrored` but do we need to store `isQuoteEscapedToken`? can we just keep appending to the token if quote are `mirrored` and that it?

Reply:
yes, it seems this approach can be improved
I'll think a little about it

Reply:
implemented a new version of the parsing algorithm